### PR TITLE
imxrt : Fix kernel debug logs

### PR DESCRIPTION
**Pull request by kishore-sn**
Kernel debug logs are not coming since some implementation for lldbg and dbg is missing. 
In this PR, I have done following:
- Fix implementation of lldbg
- Redirect dbg to lldbg (This is temporary and will be reverted when dbg implementation in imxrt is fixed)
- Enabled CONFIG_DEBUG and CONFIG_DEBUG_ERROR

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [32d40db2d3bf765351eeeb6af873b60bd944d54f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [32d40db2d3bf765351eeeb6af873b60bd944d54f] - Code Rule Check OK.

File: os/include/debug.h [hunk 1]
```diff
@@ -153,8 +153,14 @@ Once LOGM is approved, each module should have its own index
  * @details @b #include <debug.h>
  * @since TizenRT v1.0
  */
```
Comment (sangam-swami) [inline os/include/debug.h]: Can you please make separate commit for this temporary change, we can revert after the issue fix

Comment (kishore-sn) [reply os/include/debug.h]: Done.

**Comment by seinfra** [comment]
Target : [a09895e7a85af519aa3d06c2028dc04d71bd5ba4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a09895e7a85af519aa3d06c2028dc04d71bd5ba4] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_lowputc.h [hunk 2]
```diff
@@ -68,6 +68,10 @@
 #include "chip.h"
 #include "imxrt_config.h"
 
+#ifdef CONFIG_DEBUG
+#define CONFIG_DEBUG_FEATURES
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_lowputc.h]: Could you let me know the purpose of new definition?
I think we need to use `CONFIG_DEBUG`.

File: os/include/debug.h [hunk 3]
```diff
@@ -190,8 +196,13 @@ Once LOGM is approved, each module should have its own index
...
+#ifdef CONFIG_ARCH_CHIP_IMXRT
```
Comment (sunghan-chang) [inline os/include/debug.h]: I don't want to add board-specific temporary feature in common standard header.
How about redefine it in arch header?
If there is a common header which almost imxrt source codes includes, we can add below
```
#undef llwdbg
#define llwdbg wdbg
```

**Comment by sunghan-chang** [comment]
@kishore-sn
Your last change, `imxrt1050-evk/nxp_demo: Enable kernel debug logs`, is not a kernel-specific change.
Logs of all modules are enabled.

**Comment by sunghan-chang** [comment]
@kishore-sn Could you update this?

**Comment by kishore-sn** [comment]
> t1050-evk/nxp_

I changed the commit message.

File: os/arch/arm/src/imxrt/imxrt_lowputc.h (hunk 2 above)
Comment (kishore-sn) [reply os/arch/arm/src/imxrt/imxrt_lowputc.h]: True. I will change it.

File: os/include/debug.h (hunk 3 above)
Comment (kishore-sn) [reply os/include/debug.h]: I understand your concern. But, it is not possible to redefine in arch specific header. In that case our change will reflect only in arch specific files. As you know, dbg is not working properly and so, where ever dbg is used in entire TizenRT code, I want it to be replaced with lldbg. 

We will try to find a fix for the dbg issue. And revert this change as soon as possible.

**Comment by seinfra** [comment]
Target : [ea68d0b77aabaa4473a2d83ccd31180c2463b21e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ea68d0b77aabaa4473a2d83ccd31180c2463b21e] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_serial.c [hunk 4]
```diff
@@ -1520,12 +1528,11 @@ int up_putc(int ch)
...
 	if (ch == '\n') {
 		/* Add CR */
-
-		up_lowputc('\r');
+		imxrt_lowputc('\r');
 	}
 
-	up_lowputc(ch);
-	imxrt_restoreuartint(priv, intset);
+	imxrt_lowputc(ch);
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_serial.c]: Here is not in CONFIG_DEBUG conditional. But at above, definition of this function is in it. When debug is disabled, is it ok?

File: os/include/debug.h (hunk 3 above)
Comment (sunghan-chang) [reply os/include/debug.h]: ok. Could you add a comment to say this is a temporary fix?

Comment (kishore-sn) [reply os/include/debug.h]: Done.

File: os/arch/arm/src/imxrt/imxrt_serial.c (hunk 4 above)
Comment (kishore-sn) [reply os/arch/arm/src/imxrt/imxrt_serial.c]: We have this code in imxrt_lowputc.h. So it will be ok if CONFIG_DEBUG is disabled.

`#if defined(HAVE_LPUART_DEVICE) && defined(CONFIG_DEBUG)
'void imxrt_lowputc(int ch);
'#else
'#define imxrt_lowputc(ch)
'#endif
`

Comment (kishore-sn) [reply os/arch/arm/src/imxrt/imxrt_serial.c]: We have this code in imxrt_lowputc.h. So it will be ok if CONFIG_DEBUG is disabled.

#if defined(HAVE_LPUART_DEVICE) && defined(CONFIG_DEBUG)
void imxrt_lowputc(int ch);
#else
#define imxrt_lowputc(ch)
#endif

**Comment by seinfra** [comment]
Target : [0b4bf76081dbd48076c2f7ef87ca05ec0f061dd9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0b4bf76081dbd48076c2f7ef87ca05ec0f061dd9] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_serial.c (hunk 4 above)
Comment (sunghan-chang) [reply os/arch/arm/src/imxrt/imxrt_serial.c]: In compilation point of view, it's ok. But is there any reason to block it with debug disabled?

Comment (kishore-sn) [reply os/arch/arm/src/imxrt/imxrt_serial.c]: I dont think it needs to depend on CONFIG_DEBUG. I will remove the condition for CONFIG_DEBUG and keep the check for HAVE_LPUART_DEVICE.
